pkg/resources/grant_application_role_acceptance_test.go

```diff
@@ -0,0 +1,133 @@
+				),
+			},
+			// test import
+			{
+				ConfigDirectory:   config.StaticDirectory("testdata/TestAcc_GrantApplicationRole/account_role"),
+				ConfigVariables:   m(),
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAcc_GrantApplicationRole_application(t *testing.T) {
+	applicationName := "int_test_app"
+	applicationRoleName := "app_role_1"
+	applicationRoleNameFullyQualified := fmt.Sprintf("\"%s\".\"%s\"", applicationName, applicationRoleName)
+	secondApplicationName := "int_test_other_app"
+	resourceName := "snowflake_grant_application_role.g"
+
+	m := func() map[string]config.Variable {
+		return map[string]config.Variable{
+			"name":             config.StringVariable(applicationRoleNameFullyQualified),
+			"application_name": config.StringVariable(secondApplicationName),
+		}
+	}
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		PreCheck:                 func() { acc.TestAccPreCheck(t) },
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: testAccCheckGrantApplicationRoleDestroy,
+		Steps: []resource.TestStep{
+			{
+				ConfigDirectory: config.StaticDirectory("testdata/TestAcc_GrantApplicationRole/application"),
+				ConfigVariables: m(),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", applicationRoleNameFullyQualified),
+					resource.TestCheckResourceAttr(resourceName, "application_name", secondApplicationName),
+					resource.TestCheckResourceAttr(resourceName, "id", fmt.Sprintf(`"%v"."%v"|APPLICATION|"%v"`, applicationName, applicationRoleName, secondApplicationName)),
+				),
+			},
+			// test import
+			{
+				ConfigDirectory:   config.StaticDirectory("testdata/TestAcc_GrantApplicationRole/application"),
+				ConfigVariables:   m(),
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckGrantApplicationRoleDestroy(s *terraform.State) error {
+	client := acc.TestAccProvider.Meta().(*provider.Context).Client
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "snowflake_grant_application_role" {
+			continue
+		}
+		ctx := context.Background()
+		id := rs.Primary.ID
+		ids := strings.Split(id, "|")
+		applicationRoleName := ids[0]
+		objectType := ids[1]
+		parentRoleName := ids[2]
+		grants, err := client.Grants.Show(ctx, &sdk.ShowGrantOptions{
+			Of: &sdk.ShowGrantsOf{
+				ApplicationRole: sdk.NewDatabaseObjectIdentifierFromFullyQualifiedName(applicationRoleName),
+			},
+		})
+		if err != nil {
+			continue
+		}
+		for _, grant := range grants {
+			if grant.GrantedTo == sdk.ObjectType(objectType) {
+				if grant.GranteeName.FullyQualifiedName() == parentRoleName {
```

Comment:
Won't this be a problem? `parentRoleName` is saved with quotes trimmed and saved to the state and now it's compared to the identifier with quotes (`FullyQualifiedName`)

Reply:
`parentRoleName` is not trimmed now, so I think this should be OK.